### PR TITLE
Fix photo capture flow

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Save actual size photo instead of thumbnail when capturing photo.

@gino-m  PTAL?

### CODE REVIEW

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java [hunk 1]
```diff
@@ -333,35 +344,35 @@ public void onSelectPhotoResult(Uri uri) {
...
       Timber.v("Select photo result returned");
     } catch (IOException e) {
       Timber.e(e, "Error getting photo returned by camera");
     }
   }
 
-  public void onCapturePhotoResult(Bitmap thumbnail) {
-    if (thumbnail == null) {
-      Timber.e("onCapturePhotoResult called with null thumbnail");
+  public void onCapturePhotoResult(Boolean result) {
```
Comment (gino-m) [inline gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java:356]: Boolean -> boolean

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java [hunk 2]
```diff
@@ -333,35 +344,35 @@ public void onSelectPhotoResult(Uri uri) {
...
+    if (result == null || !result) {
+      Timber.e("onCapturePhotoResult called with false result");
```
Comment (gino-m) [inline gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java:358]: Is a false result necessarily an error, or could it also mean the user dismissed the capture flow? Please add a comment explaining what `result` refers too.

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java [hunk 3]
```diff
@@ -333,35 +344,35 @@ public void onSelectPhotoResult(Uri uri) {
...
+      // TODO: Cleanup created file if it exists.
       return;
     }
-    onPhotoResult(thumbnail);
+    onPhotoResult(
+        PhotoResult.createCaptureResult(
+            requireNonNull(fieldWaitingForPhoto), requireNonNull(capturedPhotoPath)));
```
Comment (gino-m) [inline gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java:364]: `requireNonNull` will crash the app if the value is null. Can we check these and log and error instead?

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java [hunk 4]
```diff
@@ -379,16 +390,37 @@ public void clearPhoto(String fieldId) {
...
+      return getBitmap().isEmpty() && getPath().isEmpty();
+    }
+
     public void setHandled(boolean handled) {
       isHandled = handled;
     }
 
-    static PhotoResult create(String fieldId, Optional<Bitmap> bitmap) {
-      return new AutoValue_EditObservationViewModel_PhotoResult(fieldId, bitmap);
+    static PhotoResult createEmptyResult(String fieldId) {
```
Comment (gino-m) [inline gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java:411]: +1 to the static constructor methods

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java [hunk 5]
```diff
@@ -120,6 +121,7 @@
   private final Observable<SaveResult> saveResults;
 
   @Nullable private String fieldWaitingForPhoto;
+  @Nullable private String capturedPhotoPath;
```
Comment (gino-m) [inline gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java:124]: Why is this needed in the view model and stored across re-creation events?

**Review by gino-m — CHANGES_REQUESTED** [review]

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java (hunk 1 above)
Comment (shobhitagarwal1612) [reply gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java]: Done

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java (hunk 2 above)
Comment (shobhitagarwal1612) [reply gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java]: Done

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java (hunk 3 above)
Comment (shobhitagarwal1612) [reply gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java]: Catching NPE in catch clause.

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java (hunk 5 above)
Comment (shobhitagarwal1612) [reply gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java:124]: For photo capture with full size, we don't get a URI. While creating the intent, we generate and pass the file URI. The result of photo capture is of type boolean. `true` refers to that the photo was saved to the passed uri correctly.

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java [hunk 6]
```diff
@@ -340,28 +340,32 @@ public void setCapturedPhotoPath(@Nullable String photoUri) {
 
   public void onSelectPhotoResult(Uri uri) {
     if (uri == null) {
-      Timber.e("onSelectPhotoResult called with null uri");
+      Timber.e("Select photo failed or canceled");
```
Comment (gino-m) [inline gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java:343]: `Timber.v`?

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java [hunk 7]
```diff
@@ -340,28 +340,32 @@ public void setCapturedPhotoPath(@Nullable String photoUri) {
...
+      Timber.e(e, "Error getting photo selected from storage");
     }
   }
 
-  public void onCapturePhotoResult(Boolean result) {
-    if (result == null || !result) {
-      Timber.e("onCapturePhotoResult called with false result");
+  public void onCapturePhotoResult(boolean result) {
+    if (!result) {
+      Timber.e("Capture photo failed or canceled");
```
Comment (gino-m) [inline gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java:358]: Timber.v

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java (hunk 5 above)
Comment (gino-m) [reply gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java:124]: So we don't get the URI back from the intent, but we need it in order to serialize? Is it the URL of the full-sized image in local storage? Please add a comment here to clarify if possible.

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java (hunk 7 above)
Comment (shobhitagarwal1612) [reply gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java]: Done

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java (hunk 6 above)
Comment (shobhitagarwal1612) [reply gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java]: Done

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java (hunk 5 above)
Comment (shobhitagarwal1612) [reply gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java:135]: Done

**Review by gino-m — APPROVED** [review]
Awesome, thanks for the fix!
